test/e2e/testsuites/dynamically_provisioned_volume_snapshot_tester.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+   http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testsuites
+
+import (
+	"sigs.k8s.io/azuredisk-csi-driver/test/e2e/driver"
+
+	"github.com/onsi/ginkgo"
+	v1 "k8s.io/api/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	restclientset "k8s.io/client-go/rest"
+)
+
+// DynamicallyProvisionedVolumeSnapshotTest will provision required StorageClass(es),VolumeSnapshotClass(es), PVC(s) and Pod(s)
+// Waiting for the PV provisioner to create a new PV
+// Testing if the Pod(s) can write and read to mounted volumes
+// Create a snapshot, validate the data is still on the disk, and then write and read to it again
+// And finally delete the snapshot
+// This test only supports a single volume
+type DynamicallyProvisionedVolumeSnapshotTest struct {
+	CSIDriver       driver.PVTestDriver
+	Pod             PodDetails
+	PodWithSnapshot PodDetails
+}
+
+func (t *DynamicallyProvisionedVolumeSnapshotTest) Run(client clientset.Interface, restclient restclientset.Interface, namespace *v1.Namespace) {
+	tpod := NewTestPod(client, namespace, t.Pod.Cmd)
```

Comment:
just want to make sure that this test case is not in the in-tree e2e tests, right? @chewong

Reply:
Thats right. https://github.com/kubernetes-sigs/azuredisk-csi-driver/commit/fd7ad22e1f8421289758f32e604b58f478e6c57b skid both snapshot and cloning test cases.